build/frontend/Dockerfile

```diff
@@ -18,25 +18,59 @@
 #   SPDX-License-Identifier: Apache-2.0
 #   ********************************************************************************
 
+# stage1 as builder
+FROM node:lts-alpine as builder
+
+WORKDIR /app
+
+# Copy the package.json and install dependencies
+COPY package*.json ./
+
+RUN npm install -g npm
+
+#RUN npm install
+RUN npm install --legacy-peer-deps
+
+# Copy rest of the files
+COPY . .
+
+# Build the project
+RUN npm run build
 
 
-# Base image
-FROM node:16.15.1
+FROM nginxinc/nginx-unprivileged:stable-alpine
+
+ARG REPO_COMMIT_ID='REPO_COMMIT_ID'
+ARG REPO_ENDPOINT_URL='REPO_ENDPOINT_URL'
+ENV REPO_COMMIT_ID=${REPO_COMMIT_ID}
+ENV REPO_ENDPOINT_URL=${REPO_ENDPOINT_URL}
+
+USER root
+
+RUN addgroup -g 3000 appgroup \
+	&& adduser -u 10000 -g 3000 -h /home/appuser -D appuser
+
+COPY ./entrypoint.sh /entrypoint.sh
 
-# Set the working directory
 WORKDIR /app
 
-# Copy the app source code
-COPY ../../demand-capacity-mgmt-frontend .
+COPY ./.nginx/nginx.conf /etc/nginx/conf.d/default.conf
+COPY --from=builder /app/dist /usr/share/nginx/html
+
+HEALTHCHECK NONE
+
+# add permissions for a user
+RUN chown -R 10000:3000 /app && chmod -R 775 /app/
+RUN chown 10000:3000 /entrypoint.sh && chmod -R 775 /entrypoint.sh
 
-# Install dependencies
-RUN yarn install
+# Install bash for env variables inject script
+RUN apk update && apk add --no-cache bash
```

Comment:
As stated in our release guidelines, We must not alter the base images of our published images.